File: roscpp_basic_pub/src/helloworld_subscriber.cpp
#include <functional>
#include <memory>

#include "rclcpp/rclcpp.hpp"
#include "std_msgs/msg/string.hpp"

using std::placeholders::_1;

void subscribe_topic_message(const std_msgs::msg::String::SharedPtr msg)
{
  RCLCPP_INFO(rclcpp::get_logger("test"), "Received message: '%s'", msg->data.c_str());
}

int main(int argc, char* argv[])
{
  rclcpp::init(argc, argv);
  auto node = std::make_shared<rclcpp::Node>("Helloworld_subscriber");
  rclcpp::Subscription<std_msgs::msg::String>::SharedPtr helloworld_subscriber;
  auto qos_profile = rclcpp::QoS(rclcpp::KeepLast(10));
  helloworld_subscriber = node->create_subscription<std_msgs::msg::String>(
    "helloworld",
    qos_profile,
    subscribe_topic_message);
  rclcpp::spin(node);
  rclcpp::shutdown();
  return 0;
}

File: rospy_basic_pub/rospy_basic_pub/helloworld_subscriber.py
import rclpy
from rclpy.node import Node
from rclpy.qos import QoSProfile
from std_msgs.msg import String

def subscribe_topic_message(self, msg):
    Node.get_logger('test').info('Received message: {0}'.format(msg.data))


def main(args=None):
    rclpy.init(args=args)
    node = Node('Helloworld_subscriber')
    qos_profile = QoSProfile(depth=10)
    helloworld_subscriber = node.create_subscription(
      String,
      'helloworld',
      subscribe_topic_message,
      qos_profile)
    try:
        rclpy.spin(node)
    except KeyboardInterrupt:
        node.get_logger().info('Keyboard Interrupt (SIGINT)')
    finally:
        node.destroy_node()
        rclpy.shutdown()

if __name__=='__main__':
  main()